graphics/cursorman.cpp

```diff
@@ -288,8 +288,8 @@ CursorManager::Cursor::Cursor(const Surface &surf, int hotspotX, int hotspotY, u
 #endif
 	_size = surf.w * surf.h * surf.format.bytesPerPixel;
 
-	// make sure that the width == pitch
-	_surf.init(surf.w, surf.h, surf.w, new byte[_size], surf.format);
```

Comment:
All of this looks like `Surface::copyFrom`.
In this case, a `_surf.free()` must be called inside the destructor.

Reply:
Indeed. The `new[]` statement is crucial. Had we used `create`, linearity of the buffer wouldn't be guaranteed. So we must to `delete[]` it and not `free` it.

Reply:
> Indeed. The `new[]` statement is crucial. Had we used `create`, linearity of the buffer wouldn't be guaranteed.

I don't understand what you mean by this.
What `new[]` brings over `malloc` used in `create` which prevents us to use `create` instead of these lines?

Reply:
We can't make assumptions of what `create` does. I just happen to know because I was to one who changed its (malloc) behaviour for my platform: https://github.com/scummvm/scummvm/blob/master/graphics/blit-atari.cpp#L101